File: jscomp/test/const_test.js
'use strict';


function f(x) {
  return x;
}

function ff(x) {
  return x;
}

function fff(x) {
  var x$1 = {
    TAG: /* A */0,
    _0: x
  };
  switch (x$1.TAG | 0) {
    case /* A */0 :
        return x;
    case /* B */1 :
        return 1;
    case /* C */2 :
        return 2;
    
  }
}

function h(x) {
  switch (x) {
    case "A" :
        return 0;
    case "B" :
        return 1;
    case "C" :
        return 2;
    
  }
}

function hh(param) {
  return 3;
}

var g = h("A");

exports.f = f;
exports.ff = ff;
exports.fff = fff;
exports.h = h;
exports.hh = hh;
exports.g = g;
/* g Not a pure module */

File: jscomp/test/const_test.ml
let f x = if true then x else x - 1
let ff x = match Some x with Some _ -> x | None -> 0

type t = A of int | B of int | C of int

let fff x = match A x with A x -> x | B _ -> 1 | C _ -> 2
let h x = match x with `A -> 0 | `B -> 1 | `C -> 2
let hh () = match "x" with "y" -> 1 | "z" -> 2 | "x" -> 3 | _ -> 4
let g = h `A
